workspaces/liferay-sample-workspace/client-extensions/liferay-sample-editor-config-contributor/src/index.ts

```diff
@@ -7,39 +7,84 @@ import {
 	EditorConfigTransformer,
 	EditorTransformer,
 } from '@liferay/js-api/editor';
+import {ButtonView, Fullscreen, Plugin} from 'ckeditor5';
 
 const editorConfigTransformer: EditorConfigTransformer<any> = (config) => {
 
 	// CKEditor 5
 
 	if (config?.editorType === 'ckeditor5') {
-		const separator = '|';
+		class HelloWorld extends Plugin {
+			init() {
+				const editor = this.editor;
+
+				editor.ui.componentFactory.add('helloworld', () => {
+					const button = new ButtonView();
+
+					button.set({
+						label: 'Hello',
+						withText: true,
+					});
+
+					button.on('execute', () => {
+						editor.model.change((writer) => {
+							editor.model.insertContent(
+								writer.createText('Hello World ')
+							);
+						});
+					});
+
+					return button;
+				});
+			}
+		}
+
 		const toolbar = [
+
+			// If plugin is in advanced preset, we can just add a toolabar entry.
+
 			'AccessibilityHelp',
-			separator,
 			'undo',
 			'redo',
-			separator,
+			'alignment',
 			{
+
+				// A dropdown with custom icon.
+
 				icon: 'text',
 				items: ['bold', 'italic', 'underline'],
-				label: 'A dropdown with a custom icon',
+				label: 'Text formatting',
 			},
-			separator,
-			'alignment',
-			separator,
 			{
 
-				// This dropdown has the icon disabled and a text label instead.
+				// A dropdown with text label instead of icon.
 
 				icon: false,
 				items: ['bulletedList', 'numberedList'],
 				label: 'Lists',
 			},
+
+			// An official plugin that was added as extra in `frontend-editor-ckeditor-sample-web`.
+
+			'bookmark',
+
+			// A custom plugin that was added in `frontend-editor-ckeditor-sample-web`.
+
+			'timestamp',
+
+			// An official plugin not in advanced preset. In addition to code here,
+			// there must be a matching symbol in `frontend-editor-ckeditor-web/node-scripts.config.js`.
+
+			'fullscreen',
```

Comment:
We may have to accept this as a product limitation, at least for now: You cannot add a new official plugin just from CX, there must be a symbol entry in DXP. I believe that fixing this would also mean fixing it in DXP. There is an ugly solution for this too, to simply add all existing symbols to `node-scripts.config.js`. That may be quite lengthy and tedious task.

cc @herrardo

Reply:
I tried something like this, it didn't work, but I may have missed something. However, I don't see how it could work. Because of importmap, we end up using `ckeditor.js` from `frontend-editor-ckeditor-web`. It does not allow second instance of CKE. The only way I can see for this to work is to copy official plugin code, and make it basically a custom plugin that happens to be the same as official one. If a customer wishes to do something like that, I suppose they can, but I don't think we should promote it :sweat_smile:

Reply:
Yeah, makes sense. If you bundle the plugin in the sample most probably that pulls another instance of ckeditor into the sample which collides with the one in DXP at runtime.